Mathlib/AlgebraicGeometry/Morphisms/Flat.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.AlgebraicGeometry.Morphisms.RingHomProperties
+import Mathlib.RingTheory.RingHom.Flat
+
+/-!
+# Flat morphisms
+
+A morphism of schemes `f : X ⟶ Y` is flat if for each affine `U ⊆ Y` and
+`V ⊆ f ⁻¹' U`, the induced map `Γ(Y, U) ⟶ Γ(X, V)` is flat.
+
+We show that this property is local, and are stable under compositions and base change.
+
+-/
+
+noncomputable section
+
+open CategoryTheory CategoryTheory.Limits Opposite TopologicalSpace
+
+universe v u
+
+namespace AlgebraicGeometry
+
+variable {X Y : Scheme.{u}} (f : X ⟶ Y)
+
+/-- A morphism of schemes `f : X ⟶ Y` is flat if for each affine `U ⊆ Y` and
+`V ⊆ f ⁻¹' U`, the induced map `Γ(Y, U) ⟶ Γ(X, V)` is flat.
+-/
```

Comment:
Can you point to `iff_flat_stalkMap` here, as that is the more common definition in the literature? (also in the module docstring)